src/data/markdown/docs/02 javascript api/10 k6-data/10-SharedArray.md

```diff
@@ -0,0 +1,44 @@
+---
+title: SharedArray
+---
+
+`SharedArray` is an array-like object that shares the underlying memory between VUs. It constructor
+takes a name for the `SharedArray` and a function which needs to return an array object itself. The
+function will be executed only once and it's result will be then be saved in memory once and copies
+of the elements will be given when requested. The name is needed as VUs are completely separate js
+VMs and k6 needs some way to identify the `SharedArray`s that it needs to return. 
+
+This does mean that you can have multiple such ones and even only load some of them for given VUs, although that is
+unlikely to have any performance benefit.
+
+Everything about `SharedArray` is read-only once it is constructed, so it is not possible to
+communicate between VUs using it.
+
+Supported operations include:
+1. getting the number of elements with `length`
+2. getting an element by it's index using the normal syntax `array[index]`
+3. using `for-of` loops
+
+Which means that for the most part if you currently have an array data structure that you want to
+take less memory you can just wrap it in `SharedArray` and it should work for most cases.
+
+### Examples
+
+<div class="code-group" data-props='{"labels": ["Simple example"], "lineNumbers": [true]}'>
+
+```javascript
+import { SharedArray } from "k6/data";
+
+var data = new SharedArray("some name", function() {
+    // here you can open files, and then do additional processing on them or just generate the data dynamically 
+    var f = JSON.parse(open("./somefile.json")).;
+    return f; // f must be an array
+});
+
+export default () => {
+    var element = data[Math.floor(Math.random() * data.length)]
+    // do something with element
```

Comment:
Let's include a more complex example showing the common use case of splitting the array among N VUs.

Reply:
Do you mean the csv example from https://k6.io/docs/examples/data-parameterization#handling-bigger-data-files (which I need to update as well) ?  Because that was a hack that is no longer needed, which the whole point was to only get parts of the array in each VU so that you don't load most of it. There is no need for that, now. Arguably having them split will be *worse* not better, so I will need to update the linked page, to specifically recommend not doing that.

Reply:
Sure, that specific example is not needed anymore, and we should just remove it in favor of this. But most people will want to get a subset of the array anyway and not a random element as shown in this example, in which case `__VU` would still be relevant. So showing that particular real-world scenario would be useful.

Reply:
I have never seen people wanting a subset of the array per iteration. They might want 1-2 elements from it, but I at least have not seen someone asking for a consecutive subset of the array. 

Also, this is not specific to the SharedArray, so I am kind of against having 1 more example.